Makefile

```diff
@@ -76,12 +76,15 @@ $(OUTPUT_DIRNAME)/$(DOC_FILENAME).html: $(DOC_FILES) $(FIGURE_FILES)
 	ls -sh $(shell readlink -f $@)
 endif
 
-validate-examples:
+validate-examples: schema/fs.go
 	go test -run TestValidate ./schema
 
-schema-fs:
+schema/fs.go: $(wildcard schema/*.json) schema/gen.go
```

Comment:
> the schema-fs ought to be commited when the JSON schema changes.

But that doesn't always happen (it didn't in #411).  Making sure it *does* always happen was what I was trying to do with #530.  I'd thought the approach you were aiming for here was to always rebuild in CI so it didn't matter if the committed copy was stale.